### PR TITLE
docs: Consolidate arm64 doc into README/building.md/uefi.md

### PR DESCRIPTION
**Pull request by MrXinWang**
Fixes: https://github.com/cloud-hypervisor/cloud-hypervisor/issues/4984

Currently marking this PR as a draft because @michael2012z will double check the Arm related commands in the doc to make sure they still work. But comments are welcomed :) @rbradford @sboeuf.

Also I split the PR to several commits for easier review. Feel free to ask a squash if it is needed when committing.

### CODE REVIEW

**Comment by michael2012z** [comment]
Thanks, @MrXinWang . Yes, it's good to cross-verify the commands by different persons.

But it's a running week to me. I need to defer testing the new commands 1 or 2 days.

**Review by michael2012z — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -69,10 +69,23 @@ Cloud Hypervisor supports `64-bit Linux` and Windows 10/Windows Server 2019.
...
-to the
-[AArch64 documentation](docs/arm64.md).
+The following sections describe how to build and run Cloud Hypervisor.
+
+## Hardware Requirements for AArch64
+
+- AArch64 servers (recommended) or development boards equipped with the GICv3
+interrupt controller.
+
+- On development boards that have constrained RAM resources, if the creation of
```
[the remark this answers is not in the document]
Comment (MrXinWang) [reply README.md:79]: Done

File: README.md [hunk 2]
```diff
@@ -105,13 +118,10 @@ do not wish to use the pre-built binaries.
 
 ## Booting Linux
 
-The instructions below are for the `x86-64` platform. For `AArch64` please see
-the [AArch64 specific documentation](docs/arm64.md).
-
-Cloud Hypervisor supports direct kernel boot (if the kernel is built with PVH
-support) or booting via a firmware (either [Rust Hypervisor
+Cloud Hypervisor supports direct kernel boot (the x86_64 kernel requires the kernel
```
[the remark this answers is not in the document]
Comment (MrXinWang) [reply README.md]: Done

File: README.md [hunk 3]
```diff
@@ -173,39 +183,51 @@ $ ./cloud-hypervisor \
...
+$ cp linux-config-aarch64 .config # AArch64
+# Do native build of the x86_64 kernel
 $ KCFLAGS="-Wa,-mx86-used-note=no" make bzImage -j `nproc`
+# Do native build of the AArch64 kernel
+$ make -j `nproc`
 $ popd
 ```
 
-The `vmlinux` kernel image will then be located at
+For x86_64, The `vmlinux` kernel image will then be located at
```
[the remark this answers is not in the document]
Comment (MrXinWang) [reply README.md]: done

File: README.md [hunk 4]
```diff
@@ -173,39 +183,51 @@ $ ./cloud-hypervisor \
...
 `linux-cloud-hypervisor/arch/x86/boot/compressed/vmlinux.bin`.
+For AArch64, The `Image` kernel image will then be located at
```
[the remark this answers is not in the document]
Comment (MrXinWang) [reply README.md]: done

**Review by michael2012z — APPROVED** [review]

**Review by likebreath — APPROVED** [review]

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rbradford) [reply README.md:79]: I'm not sure this is still a valid concern. The issue with `MAP_PRIVATE` and `memfd_create` was the cause of this problem. Obviously if you try and create VMs with a total memory greater than your available RAM you will have issues but those issues are not architecture dependent. I  think this whole paragraph can be removed.

[the remark this answers is not in the document]
Comment (rbradford) [reply README.md:79]: @MrXinWang I think you accidentally resolved this thread. I don't think the paragraph talking about swap makes sense and it's not an aarch64 specific issue anyway.

[the remark this answers is not in the document]
Comment (MrXinWang) [reply README.md]: Hi @rbradford 

> I think you accidentally resolved this thread.

I am really sorry.

> I don't think the paragraph talking about swap makes sense and it's not an aarch64 specific issue anyway.

I think Michael is also in favor of dropping this so I removed this whole paragraph. Thanks for your review.

**Review by rbradford — APPROVED** [review]

**Comment by rbradford** [comment]
Thank you!

**Comment by rbradford** [comment]
@MrXinWang Perhaps you could update the website to match: https://github.com/cloud-hypervisor/cloud-hypervisor.github.io ?

**Comment by MrXinWang** [comment]
Hi @rbradford, sure, I can have a look.

**Pull request merged**